api/authenticator.py

```diff
@@ -639,7 +658,7 @@ def create_authentication_document(self):
 
         library_name = self.library_name or unicode(_("Library"))
         doc = OPDSAuthenticationDocument.fill_in(
-            base_opds_document, list(self.providers),
+            base_opds_document, list(self.providers), self._db,
```

Comment:
Would it be better if you passed in the Library object or its short name directly?